### PR TITLE
Adds Geo gen2

### PR DESCRIPTION
**Pull request by ykethan**
#### Description of changes:

Adds geo CDK examples and moves geo docs

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/amazon-location-sdk/index.mdx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+If you need functionality in the AWSLocation framework used by the Amplify Geo category that isn't available, we provide an escape hatch so you can reference it directly.
+
+<Callout warning>
+
+**Note:** While the Amplify Library for Swift is production ready, please note that the underlying AWS SDK for Swift is currently in Developer Preview, and is not yet intended for production workloads. [Here is additional reading material](https://github.com/awslabs/aws-sdk-swift/blob/main/Sources/Core/AWSSDKForSwift/Documentation.docc/stability.md) on the stability of the SDK
+
+</Callout>
+
+Note: If you provisioned your Geo resources via Amplify CLI, then the IAM policy will be specifically scoped to only allow actions required by the library. Please [adjust your authorization permissions](/[platform]/build-a-backend/more-features/geo/existing-resources/) accordingly for your escape hatch use-cases.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/add-aws-services/geo/amazon-location-sdk/index.mdx:234]: this doesn't make sense anymore in Gen 2, right?

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/existing-resources/index.mdx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Now you can proceed to [displaying a map](/gen1/[platform]/build-a-backend/more-features/geo/maps/) or [adding location search](/gen1/[platform]/build-a-backend/more-features/geo/location-search/) to your app.
+</InlineFilter>
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/geo/existing-resources/index.mdx:172]: Should these links be pointing to gen1? Just wanted to verify

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/existing-resources/index.mdx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+
+<InlineFilter filters={['swift']}>
+## In your app configuration
+
+Amplify Geo is dependent on your Amplify Auth category. If it is not already configured, then you will need to [configure it](/gen1/[platform]/build-a-backend/auth/existing-resources) as well.
+
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/geo/existing-resources/index.mdx:179]: Just wanted to verify, should this link be pointing to gen1?

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/amazon-location-sdk/index.mdx (hunk 1 above)
Comment (ykethan) [reply src/pages/[platform]/build-a-backend/add-aws-services/geo/amazon-location-sdk/index.mdx]: thank you removed it

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/existing-resources/index.mdx (hunk 3 above)
Comment (ykethan) [reply src/pages/[platform]/build-a-backend/add-aws-services/geo/existing-resources/index.mdx:179]: updated the links

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/set-up-geo/index.mdx [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+});
+
+// apply the policy to the authenticated and unauthenticated roles
+backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(myGeoPolicy);
+backend.auth.resources.unauthenticatedUserIamRole.attachInlinePolicy(myGeoPolicy);
+
+// patch the map resource to the expected output configuration
+backend.addOutput({
+  geo: {
+    aws_region: Stack.of(geoStack).region,
```
Comment (edwardfoyle) [inline src/pages/[platform]/build-a-backend/add-aws-services/geo/set-up-geo/index.mdx:87]: nit: geoStack should already be of type Stack

Comment (ykethan) [reply src/pages/[platform]/build-a-backend/add-aws-services/geo/set-up-geo/index.mdx]: thanks ed

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/geofences/index.mdx [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+
+  </Block>
+
+</BlockSwitcher>
+
+> **Note:** Ensure that your package bundler (webpack, rollup, etc) is configured to handle css files. Check out the webpack documentation [here](https://webpack.js.org/loaders/css-loader/).
+
+![Geofence Create example when selecting an area on a map](/images/geofence-create.png)
+
+![Geofence Edit example when selecting an area on a map](/images/geofence-edit.png)
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/geo/geofences/index.mdx:157]: Could we get the alt text to be a little bit more descriptive? I think mentioning what the "Create example" and the "Edit example" are supposed to show would be helpful

File: src/pages/[platform]/build-a-backend/add-aws-services/geo/google-migration/index.mdx [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+// Create the search box and link it to the UI element.
+const input = document.getElementById("pac-input");
+const searchBox = new google.maps.places.SearchBox(input);
+```
+
+  </Block>
+
+</BlockSwitcher>
+
+![A search box with starbucks as input](/images/geocoder-search-box.png)
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/geo/google-migration/index.mdx:381]: Maybe for the alt text, we could also add, `A search box with Starbucks as input and the search results from the search` ?

**Review by timngyn — APPROVED** [review]

**Review by josefaidt — APPROVED** [review]

**Pull request merged**